### PR TITLE
add list rule with service_group_id test

### PR DESCRIPTION
**Pull request by cam-d12**
Signed-off-by: Cameron Davis <cjdavis@ibm.com>

## PR summary

This PR adds CBR support for the new service_group_id parameter in rules.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  

No current behavior should be modified. Adds the further control to be able to apply rules to service groups

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

File: test/integration/context-based-restrictions.v1.test.js [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
 const { readExternalSources } = require('ibm-cloud-sdk-core');
 const { v4: uuidv4 } = require('uuid');
-const ContextBasedRestrictionsV1 = require('../../dist/context-based-restrictions/v1');
+// const ContextBasedRestrictionsV1 = require('../../dist/context-based-restrictions/v1');
+const ContextBasedRestrictionsV1 = require('@ibm-cloud/platform-services/context-based-restrictions/v1');
```
Comment (soohoowoohoo) [inline test/integration/context-based-restrictions.v1.test.js:22]: @cam-d12 did you make this change or the generator code? The dist folder should have the latest locally built code. If you need to help with how to build the code into the dist folder, try the readme or reach out and I can try to help.

File: test/integration/context-based-restrictions.v1.test.js [hunk 2]
```diff
@@ -580,6 +581,77 @@ describe('ContextBasedRestrictionsV1_integration', () => {
...
+    //list rule with service_group_id attribute
+    const listParams = {
+      accountId,
+      transactionId: uuidv4(),
+    };
+
+    const listRes = await contextBasedRestrictionsService.listRules(listParams);
+    expect(listRes).toBeDefined();
+    expect(listRes.status).toBe(200);
+    expect(listRes.result).toBeDefined();
```
Comment (soohoowoohoo) [inline test/integration/context-based-restrictions.v1.test.js:641]: I think we'll want to assert that only the 1 service group rule was returned.

File: test/integration/context-based-restrictions.v1.test.js [hunk 3]
```diff
@@ -580,6 +581,77 @@ describe('ContextBasedRestrictionsV1_integration', () => {
...
+
+    expect(res).toBeDefined();
+    expect(res.status).toBe(201);
+    expect(res.result).toBeDefined();
+
+    ruleId = res.result.id;
+    ruleEtag = res.headers.etag;
+
+    //list rule with service_group_id attribute
+    const listParams = {
```
Comment (soohoowoohoo) [inline test/integration/context-based-restrictions.v1.test.js:633]: I would expect that we'd want to pass the serviceGroupId parameter to make sure filtering by the property is working as expected?

**Review by soohoowoohoo — CHANGES_REQUESTED** [review]

**Review by soohoowoohoo — COMMENTED** [review]
LGTM

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**